service/src/main/java/bio/terra/cbas/runsets/inputs/InputGenerator.java

```diff
@@ -29,6 +41,82 @@ public class InputGenerator {
           .configure(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS, true)
           .build();
 
+  public static ParameterTypeDefinition recursivelyGetParameterType(ValueType valueType)
```

Comment:
"Recursively" is more of an implementation detail than something that affects the output, so I'd suggest renaming to just `getParameterType`.